app/helpers/application_helper.rb

```diff
@@ -260,6 +244,18 @@ def view_to_url(view, parent = nil)
     end
   end
 
+  def class_to_db(cname, parent)
+    return unless cname
+    case
```

Comment:
Yes!  That's right.  That would then change line 253 and 254 below to `when "SystemService"` where we'd have to move subcondition into that single "when" of the case statement.